### PR TITLE
Introduce the support for reading private/public keys from the content

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Implementation of the proposal https://github.com/ballerina-platform/ballerina-library/issues/6517

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

File: ballerina/private_public_key.bal [hunk 1]
```diff
@@ -170,6 +170,19 @@ public isolated function decodeRsaPrivateKeyFromKeyFile(string keyFile, string?
     'class: "io.ballerina.stdlib.crypto.nativeimpl.Decode"
 } external;
 
+# Decodes the RSA private key from the given private key content as a byte array.
```
Comment (dilanSachi) [inline ballerina/private_public_key.bal:173]: I guess we can add support for key types other than rsa in the future right?

**Comment by ayeshLK** [comment]
Shall we add the proposal to the package `docs` ?

File: ballerina/private_public_key.bal (hunk 1 above)
Comment (Bhashinee) [reply ballerina/private_public_key.bal:173]: Yes, due to time limitations, it was implemented only for the RSA type. We will be introducing this to EC as well. Created issue :- https://github.com/ballerina-platform/ballerina-library/issues/6581

File: docs/proposals/decode-keys-from-the-content.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Proposal: Introduce support to retrieve `crypto:PrivateKey` and `crypto:PublicKey` by directly using the content
+
+_Authors_: @ayeshLK  
+_Reviewers_: @shafreenAnfar @bhashinee @NipunaRanasinghe
```
Comment (ayeshLK) [inline docs/proposals/decode-keys-from-the-content.md:4]:
```suggestion
_Reviewers_: @shafreenAnfar @daneshk @bhashinee @NipunaRanasinghe  
```

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**